cdap-app-fabric/src/main/java/co/cask/cdap/internal/app/runtime/artifact/ArtifactRepository.java

```diff
@@ -133,6 +136,47 @@ public ArtifactDetail getArtifact(Id.Artifact artifactId) throws IOException, Ar
   }
 
   /**
+   * Get all application classes in the given namespace, optionally including classes from system artifacts as well.
+   * Will never return null. If no artifacts exist, an empty list is returned. Namespace existence is not checked.
+   *
+   * @param namespace the namespace to get application classes from
+   * @param includeSystem whether classes from system artifacts should be included in the results
+   * @return an unmodifiable list of application classes that belong to the given namespace
+   * @throws IOException if there as an exception reading from the meta store
+   */
+  public List<ApplicationClassSummary> getApplicationClasses(Id.Namespace namespace,
+                                                             boolean includeSystem) throws IOException {
+    List<ApplicationClassSummary> summaries = Lists.newArrayList();
+    if (includeSystem) {
+      addAppSummaries(summaries, Id.Namespace.SYSTEM);
+    }
+    addAppSummaries(summaries, namespace);
+
+    return Collections.unmodifiableList(summaries);
+  }
+
+  /**
+   * Get all application classes in the given namespace of the given class name.
+   * Will never return null. If no artifacts exist, an empty list is returned. Namespace existence is not checked.
+   *
+   * @param namespace the namespace to get application classes from
+   * @param className the application class to get
+   * @return an unmodifiable list of application classes that belong to the given namespace
+   * @throws IOException if there as an exception reading from the meta store
+   */
+  public List<ApplicationClassInfo> getApplicationClasses(Id.Namespace namespace,
```

Comment:
there is no include system because the flag only applies to the top level get call (get /classes/apps).  When you ask for a specific class (get /classes/apps/co.cask....) there is a isSystem flag that determines whether or not its a system class.